src/som/compiler/MixinBuilder.java

```diff
@@ -255,6 +255,14 @@ public void setMixinResolverSource(final SourceSection mixin) {
     mixinResolversSource = mixin;
   }
 
+  public ExpressionNode constructSuperClassResolution(final SSymbol superClass,
+      final SourceSection source) {
+    MethodBuilder def = getClassInstantiationMethodBuilder();
+    ExpressionNode selfRead = def.getSelfRead(source);
+    return SNodeFactory.createMessageSend(superClass,
+        new ExpressionNode[] {selfRead}, false, source, null, language);
+  }
+
```

Comment:
Why has this been added / changed?

Reply:
It's a refactor, extracting common code.